app/views/categories/_form.html.erb

```diff
@@ -10,12 +10,31 @@
           </ul>
         </div>
     <% end %>
-
     <div class="field">
-      <%= f.label :name %><br>
+      <%= f.label 'Category Name' %><br>
       <%= f.text_field :name %>
     </div>
+    <%i=1%>
+    <%=f.fields_for :products do |product| %>
```

Comment:
instead of doing `i=1.....` you could have used `<%=f.fields_for :products do |index,product| %>`